src/nvim/mark_extended.h

```diff
@@ -0,0 +1,65 @@
+#ifndef NVIM_EXTMARK_H
+#define NVIM_EXTMARK_H
+
+#include "nvim/mark_extended_defs.h"
+#include "nvim/lib/kbtree.h"
+#include "nvim/lib/kvec.h"
+#include "nvim/map.h"
+
+/* typedef struct ExtendedMark ExtendedMark; */
+typedef struct ExtendedMark {
+  char *name;
+  pos_T mark;
+} ExtendedMark, *ExtendedMarkPtr;
+
+#define extmark_pos_cmp(a, b) (pos_cmp((a)->mark, (b)->mark))
+
+/* prev pointer is used for accessing the previous extmark iterated over */
+/* TODO there should be a better way to get this.. maybe build it in to kbtree */
+/* exists checks if the extmark is there for a particular buffer namespace */
+#define FOR_ALL_EXTMARKS_WITH_PREV(buf) \
```

Comment:
it seems that we want `kb_itr_prev`, shouldn't be too hard to extrapolate from `kb_itr_next`.